video/out/hwdec/hwdec_vaapi.c

```diff
@@ -169,37 +103,16 @@ static int init(struct ra_hwdec *hw)
 {
     struct priv_owner *p = hw->priv;
 
+    // We call all possible inits because these will do nothing if the ra context
+    // doesn't match.
 #if HAVE_GL
-    if (ra_is_gl(hw->ra)) {
-        if (!eglGetCurrentContext())
-            return -1;
-
-        const char *exts = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS);
-        if (!exts)
-            return -1;
-
-        GL *gl = ra_gl_get(hw->ra);
-        if (!strstr(exts, "EXT_image_dma_buf_import") ||
-            !strstr(exts, "EGL_KHR_image_base") ||
-            !strstr(gl->extensions, "GL_OES_EGL_image") ||
-            !(gl->mpgl_caps & MPGL_CAP_TEX_RG))
-            return -1;
-
-        MP_VERBOSE(hw, "using VAAPI EGL interop\n");
-    }
+    if (!vaapi_gl_init(hw))
+        return -1;
 #endif
 
 #if HAVE_VULKAN
-    if (ra_pl_get(hw->ra)) {
-        const struct pl_gpu *gpu = ra_pl_get(hw->ra);
-        if (!(gpu->import_caps.tex & PL_HANDLE_DMA_BUF)) {
-            MP_VERBOSE(hw, "VAAPI Vulkan interop requires support for "
-                           "dma_buf import in Vulkan.\n");
-            return -1;
-        }
-
-        MP_VERBOSE(hw, "using VAAPI Vulkan interop\n");
-    }
+    if (!vaapi_vk_init(hw))
+        return -1;
 #endif
```

Comment:
That looks strange. What happens if both fail to init (and return true)? What happens if both succeed to init somehow? I know the latter can't currently happen, but from the perspective of this code this is a coincidence, and it could change if more backends are added.